Sources/SwiftDocC/Infrastructure/External Data/GlobalExternalSymbolResolver.swift

```diff
@@ -0,0 +1,36 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021-2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+/// A type that resolves symbol identifiers for symbols in all non-local modules.
+///
+/// Use this protocol to integrate symbol documentation content from other sources. Your implementation needs to be able to:
+///  * Resolve symbols by their precise identifier  and return the symbols reference and external entity.
+///
+/// If a symbol in a symbol graph file references a symbol in another module—for example, when the symbol conforms to a protocol from another module, when the
```

Comment:
```suggestion
/// If a symbol in a symbol graph file references a symbol in another module -- for example, when the symbol conforms to a protocol from another module, when the
```

Reply:
I replaced all em dashes with `---` in all documentation comments in https://github.com/apple/swift-docc/pull/806/commits/bfd1a96237d4a1cbaaa6d64b744abcb40afa28e1